### ISSUE TITLE
Autocompletion for named arguments in Jupyter Notebook / IPython

### ISSUE THREAD

**Issue by mlund** [pybind/pybind11#990]
## Issue description

In Jupyter Notebook or in IPython I can normally press tab inside a function to autocomplete named arguments, but I cannot make this work for the example below. I can explicitly name the arguments, i.e. `add(first=2, second=3)` but not begin typing `sec...`+tab to complete to `second`. Is there a way of achieve this?

## Reproducible example code

```.cpp
#include <pybind11/pybind11.h>
int add(int i, int j) {
    return i + j;
}
namespace py = pybind11;
PYBIND11_PLUGIN(example) {
    using namespace pybind11::literals;
    py::module m("example", "pybind11 example plugin");
    m.def("add", &add, "A function which adds two numbers", "first"_a, "second"_a);
    return m.ptr();
}
```

**Comment by dean0x7d** [comment]
Currently, IPython can't get argument names from pybind11 functions (some other tools can get this information by parsing docstrings, but it doesn't look like IPython does that). This should be resolved by #945 which would allow IPython to inspect the function signature.

**Comment by mlund** [comment]
Thanks for looking into this - I'll watch #945. Signature inspection / argument completion is in my case key for writing a python interface as it mostly eliminates the need for external documentation.
